workflow/controller/scope.go

```diff
@@ -92,47 +90,7 @@ func (s *wfScope) resolveArtifact(v string, subPath string) (*wfv1.Artifact, err
 
 		// Copy resolved artifact pointer before adding subpath
 		copyArt := valArt.DeepCopy()
-
-		locationType := copyArt.ArtifactLocation.GetType()
```

Comment:
refactored into `wfv1.ArtifactLocationType.AppendToKey`

Reply:
Very clean refactor. Tested and worked on my end + makes for cleaner artifact resolution logic + unit tests are happy without changing them.

We're good from a "subPath" perspective.